connor/itunes-node
# A little contribution

Hey, @connor. Thanks for this awesome package!

Due to Hacktoberfest 2018, I'm contributing to some packages that I use; yours is one of those.

My initial idea was only to add TypeScript (TS) typings, but since I saw it that was a really old package I've decided to do some other changes:

* [x] - Changed it to a TS project;
* [x] - Made it ES6 compatible removing the callbacks;
* [x] - Removed **ALL** dependencies -- making the package more compatible, fast and almost weightless;
    * git hooks with Husky;
    * Webpack integration;
    * tests with Jest;
    * linter;
    * .npmignore;
* [x] - Also I have added a Continuos Integration (CI) added later:
    * Travis CI
    * Codecov
    * Codeclimate
* [x] - Improved README;
* [x] - Updating docs;
* [x] - Making use of typedoc to integrate with Github pages - https://fazendaaa.github.io/itunes-node/
* [x] - Removed the examples folder, setting the tests as an "example" reference.
* [x] - Linked typings file in your package.json -- making no need to submit to @DefinitelyTyped;

TODO:
If you accept this PR:

* Set a LICENSE.
* Remove all reference in README.md to ```Fazendaaa``` and set it to ```connor``` -- a simple Regex should do it.
* Configure the Github pages in your repository options -- is easy.

This is the project that I use your package, I intend to back to it someday: https://github.com/Fazendaaa/podsearch_bot.
